src/01-dinosaur-facts.js

```diff
@@ -71,7 +135,54 @@ function getDinosaurDescription(dinosaurs, id) {}
  *  getDinosaursAliveMya(dinosaurs, 65, "unknown-key");
  *  //> ["WHQcpcOj0G"]
  */
-function getDinosaursAliveMya(dinosaurs, mya, key) {}
+
+
+function getDinosaursAliveMya(dinosaurs, mya, key) {
+
+  //returning array of dinos that were alive based on mya
+  // if key is provided, return key value
+  //else, return ID if key is not provided
+  //if dino has single value for mya, mya === value or -1 from the number in the array [mya]
+
+  let newArr = [];
+  
+  for (let dino of dinosaurs) {
+    if(dino.mya.length === 1){ // single value
+      if((dino.mya[0] >= mya && dino.mya[0]-1 <= mya) && key){ // if value is equal to or one less
+        newArr.push(dino[key]); //key is pushed if user types in one less
+      } 
+    } //two values//
+      else if(dino.mya.length === 2 && dino.mya[0] >= mya && dino.mya[1] <= mya && key) // if key included
+        newArr.push(dino[key]); // add key to array
+      }
+        // else if(dino.mya[0] >= mya && dino.mya[1] <= mya){
+        // newArr.push(dino.dinosaurId);
+    return newArr; 
```

Comment:
it looks like you are not handling the edge case of your key no being defined and having to push the dino IDs.